CustomerInformation.cs

```diff
@@ -90,70 +61,25 @@ private void btnDelete_Click(object sender, EventArgs e)
 
         private void btnClear_Click(object sender, EventArgs e)
         {
-            txtFirstName.Clear();
+         
             txtLastName.Clear();
             txtTelephone.Clear();
             txtMail.Clear();
             txtID.Clear();
-            txtPrice.Clear();
-            txtLoginDate.Text= "";
-            txtExitDate.Text= "";
+           
         }
 
         private void btnUpdate_Click(object sender, EventArgs e)
         {
-            using (var connection = new SqlConnection(ConnectionString))
-            {
-                    connection.Open();
-                    SqlCommand command = new SqlCommand("update customer set firstName= @firstName , lastName= @lastName, telephone= @telephone, mail= @mail, TC= @TC, price= @price where ID = @id", connection);
-                    command.Parameters.Add(new System.Data.SqlClient.SqlParameter("@id", SqlDbType.Int, 3) { Value = id });
-                    command.Parameters.Add(new System.Data.SqlClient.SqlParameter("@firstName", SqlDbType.VarChar, 50) { Value = txtFirstName.Text });
-                    command.Parameters.Add(new System.Data.SqlClient.SqlParameter("@lastName", SqlDbType.NVarChar, 50) { Value = txtLastName.Text });
-                    command.Parameters.Add(new System.Data.SqlClient.SqlParameter("@telephone", SqlDbType.VarChar, 11) { Value = txtTelephone.Text });
-                    command.Parameters.Add(new System.Data.SqlClient.SqlParameter("@mail", SqlDbType.VarChar, 50) { Value = txtMail.Text });
-                    command.Parameters.Add(new System.Data.SqlClient.SqlParameter("@TC", SqlDbType.VarChar, 11) { Value = ID.Text });
-                    command.Parameters.Add(new System.Data.SqlClient.SqlParameter("@price", SqlDbType.Int , 20) { Value = txtPrice.Text });
-
-                    command.ExecuteNonQuery();
-                    connection.Close();
-                    showdata();
-            }
            
         }
 
         private void btnResearch_Click(object sender, EventArgs e)
         {
-            using (var connection = new SqlConnection(ConnectionString))
-            {
-                listShowdata.Items.Clear();
-                connection.Open();
-                SqlCommand command = new SqlCommand("select ID, firstName, lastName, telephone, mail, TC, price, loginDate, exitDate from customer where ID= @researchID", connection);
-                command.Parameters.Add(new System.Data.SqlClient.SqlParameter("@researchID", SqlDbType.Int, 3) { Value = txtResearchID.Text });
-                SqlDataReader reader = command.ExecuteReader();
```

Comment:
bu kısmı DataAcces tarafına taşımak için sildim. 
o kısımda bu metodu yazarken sorguyu Id üzerinden değilde Name Surname üzerinden yazmayı düşünüyorum. çünkü Id'yi bilmiyorsak bu bilgi olmadan da  arama yapamayız. Name Surname ile sorgulayınca aynı isim ve soyad da bir çok kişi olsa bile son kayıt  en üste olacaktır böylece kolayca aranabilir de olur.
Sizce nasıl olur ? @zencifil

Reply:
Customer arama isi yapiyorsan zaten kimse id bilmeyecek. O yuzden ad soyad (ki o da tam eslesme olmak zorunda olmamali, genelde soyadi yazinca listelenir) ile arama yapmak en iyisi.